controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -428,13 +430,15 @@ public void invalidateCache(String table, String key) {
                                      List<String> items = result.getItems().stream().map(x -> new String(getArray(x.getKey()), Charsets.UTF_8))
                                                                 .collect(Collectors.toList());
                                      log.trace("get keys paginated on table {} returned items {}", tableName, items);
-                                     return new AbstractMap.SimpleEntry<>(Unpooled.wrappedBuffer(result.getState().toBytes()), items);
+                                     // if the returned token and result are empty, return the incoming token so that 
+                                     // callers can resume from that token. 
+                                     return new AbstractMap.SimpleEntry<>(getNextToken(continuationToken, result), items);
```

Comment:
What is this for?

Reply:
The promise of controller vis a vis the continuation token is that it is an artifact from which if you make a query to read next set of items, it will retrieve them from the store and if there are no more results to be found, it would return empty list but the same token.
Basically token indicates the position till which the results have been consumed. 

if they have exhausted the whole result set, this should point to the last element in the set.
however, the read all keys/entries api on segment store doesnt return the same token back, instead it returns an empty token if there are no more items to return. 
for example: if you issue `readAllKeys(empty)` and `get {List<Keys>, Token}` .. lets say all keys are read.. now if you issue readAllKeys(token), segment store returns {empty List, empty token}`
This makes sure we return the same token back instead so that any time user makes a subsequent call from the token, it gets a response from which it can continue without reading the whole result again.